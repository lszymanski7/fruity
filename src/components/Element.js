import React from 'react'
import PropTypes from 'prop-types'

const Element = ({ checked, handleCheck, id, img, name }) => {
    return (
        <div className="element">
            <label className="element__label">
                <input
                    checked={checked}
                    className="element__checkbox"
                    onChange={() => handleCheck(id)}
                    type="checkbox"
                />
                <img
                    alt={name}
                    className="element__image"
                    src={img}
                />
                <span className="element__caption">{name.toUpperCase()}</span>
            </label>
        </div>
    )
}

Element.propTypes = {
    checked: PropTypes.bool.isRequired,
    handleCheck: PropTypes.func.isRequired,
    id: PropTypes.number.isRequired,
    img: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired
}

export { Element as default }
