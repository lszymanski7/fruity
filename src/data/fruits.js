import { ICONS } from './constants'

// Fruits data
const fruits = [
    {
        id: 1,
        name: 'apple',
        img: ICONS.apple,
        checked: false
    },
    {
        id: 2,
        name: 'banana',
        img: ICONS.banana,
        checked: false
    },
    {
        id: 3,
        name: 'blackberry',
        img: ICONS.blackberry,
        checked: false
    },
    {
        id: 4,
        name: 'blueberries',
        img: ICONS.blueberries,
        checked: false
    },
    {
        id: 5,
        name: 'cherries',
        img: ICONS.cherries,
        checked: false
    },
    {
        id: 6,
        name: 'coconut',
        img: ICONS.coconut,
        checked: false
    },
    {
        id: 7,
        name: 'dragon fruit',
        img: ICONS.dragon_fruit,
        checked: false
    },
    {
        id: 8,
        name: 'fig',
        img: ICONS.fig,
        checked: false
    },
    {
        id: 9,
        name: 'grapes',
        img: ICONS.grapes,
        checked: false
    },
    {
        id: 10,
        name: 'kiwi',
        img: ICONS.kiwi,
        checked: false
    },
    {
        id: 11,
        name: 'lemon',
        img: ICONS.lemon,
        checked: false
    },
    {
        id: 12,
        name: 'lychee',
        img: ICONS.lychee,
        checked: false
    },
    {
        id: 13,
        name: 'mango',
        img: ICONS.mango,
        checked: false
    },
    {
        id: 14,
        name: 'melon',
        img: ICONS.melon,
        checked: false
    },
    {
        id: 15,
        name: 'orange',
        img: ICONS.orange,
        checked: false
    },
    {
        id: 16,
        name: 'papaya',
        img: ICONS.papaya,
        checked: false
    },
    {
        id: 17,
        name: 'peach',
        img: ICONS.peach,
        checked: false
    },
    {
        id: 18,
        name: 'pear',
        img: ICONS.pear,
        checked: false
    },
    {
        id: 19,
        name: 'pineapple',
        img: ICONS.pineapple,
        checked: false
    },
    {
        id: 20,
        name: 'plum',
        img: ICONS.plum,
        checked: false
    },
    {
        id: 21,
        name: 'pomegranate',
        img: ICONS.pomegranate,
        checked: false
    },
    {
        id: 22,
        name: 'star fruit',
        img: ICONS.star_fruit,
        checked: false
    },
    {
        id: 23,
        name: 'strawberry',
        img: ICONS.strawberry,
        checked: false
    },
    {
        id: 24,
        name: 'watermelon',
        img: ICONS.watermelon,
        checked: false
    }
]

export { fruits as default }
