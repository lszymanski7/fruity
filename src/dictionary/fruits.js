import { fruitsColor, fruitsLinear } from '../assets/icons/index'

const fruits = [{
    name: 'Apple',
    img: [
        fruitsColor['apple.png'].default,
        fruitsLinear['apple.png'].default
    ]
}, {
    name: 'Banana',
    img: [
        fruitsColor['banana.png'].default,
        fruitsLinear['banana.png'].default
    ]
}, {
    name: 'Blackberry',
    img: [
        fruitsColor['blackberry.png'].default,
        fruitsLinear['blackberry.png'].default
    ]
}, {
    name: 'Blueberries',
    img: [
        fruitsColor['blueberries.png'].default,
        fruitsLinear['blueberries.png'].default
    ]
}, {
    name: 'Cherries',
    img: [
        fruitsColor['cherries.png'].default,
        fruitsLinear['cherries.png'].default
    ]
}, {
    name: 'Coconut',
    img: [
        fruitsColor['coconut.png'].default,
        fruitsLinear['coconut.png'].default
    ]
}, {
    name: 'Dragon Fruit',
    img: [
        fruitsColor['dragon-fruit.png'].default,
        fruitsLinear['dragon-fruit.png'].default
    ]
}, {
    name: 'Fig',
    img: [
        fruitsColor['fig.png'].default,
        fruitsLinear['fig.png'].default
    ]
}, {
    name: 'Grapes',
    img: [
        fruitsColor['grapes.png'].default,
        fruitsLinear['grapes.png'].default
    ]
}, {
    name: 'Kiwi',
    img: [
        fruitsColor['kiwi.png'].default,
        fruitsLinear['kiwi.png'].default
    ]
}, {
    name: 'Lemon',
    img: [
        fruitsColor['lemon.png'].default,
        fruitsLinear['lemon.png'].default
    ]
}, {
    name: 'Lychee',
    img: [
        fruitsColor['lychee.png'].default,
        fruitsLinear['lychee.png'].default
    ]
}, {
    name: 'Mango',
    img: [
        fruitsColor['mango.png'].default,
        fruitsLinear['mango.png'].default
    ]
}, {
    name: 'Melon',
    img: [
        fruitsColor['melon.png'].default,
        fruitsLinear['melon.png'].default
    ]
}, {
    name: 'Orange',
    img: [
        fruitsColor['orange.png'].default,
        fruitsLinear['orange.png'].default
    ]
}, {
    name: 'Papaya',
    img: [
        fruitsColor['papaya.png'].default,
        fruitsLinear['papaya.png'].default
    ]
}, {
    name: 'Peach',
    img: [
        fruitsColor['peach.png'].default,
        fruitsLinear['peach.png'].default
    ]
}, {
    name: 'Pear',
    img: [
        fruitsColor['pear.png'].default,
        fruitsLinear['pear.png'].default
    ]
}, {
    name: 'Pineapple',
    img: [
        fruitsColor['pineapple.png'].default,
        fruitsLinear['pineapple.png'].default
    ]
}, {
    name: 'Plum',
    img: [
        fruitsColor['plum.png'].default,
        fruitsLinear['plum.png'].default
    ]
}, {
    name: 'Pomegranate',
    img: [
        fruitsColor['pomegranate.png'].default,
        fruitsLinear['pomegranate.png'].default
    ]
}, {
    name: 'Star Fruit',
    img: [
        fruitsColor['star-fruit.png'].default,
        fruitsLinear['star-fruit.png'].default
    ]
}, {
    name: 'Strawberry',
    img: [
        fruitsColor['strawberry.png'].default,
        fruitsLinear['strawberry.png'].default
    ]
}, {
    name: 'Watermelon',
    img: [
        fruitsColor['watermelon.png'].default,
        fruitsLinear['watermelon.png'].default
    ]
}]

export { fruits as default }