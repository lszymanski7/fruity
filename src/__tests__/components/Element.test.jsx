import React from 'react'
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import Element from '../../components/Element'
import fruits from '../../data/fruits'

const props = {
  checked: fruits[0].checked,
  handleCheck: jest.fn(),
  id: fruits[0].id,
  img: fruits[0].img,
  name: fruits[0].name
}

describe('Element', () => {
  it('should be rendered correctly.', () => {
    const { container } = render(<Element {...props} />)
    expect(container).toMatchSnapshot()
  })

  it('should handle clicking the checkbox.', async () => {
    render(<Element {...props} />)
    const checkbox = screen.getByRole('checkbox')
    await userEvent.click(checkbox)
    expect(props.handleCheck).toHaveBeenCalled()
  })

  it('should have the correct image.', () => {
    render(<Element {...props} />)
    const img = screen.getByRole('img')
    expect(img).toHaveAttribute('alt', props.name)
    expect(img).toHaveAttribute('src', props.img)
  })

  it('should have the correct caption.', () => {
    render(<Element {...props} />)
    const span = screen.getByText('APPLE')
    expect(span).toHaveTextContent(props.name.toUpperCase())
  })
})
