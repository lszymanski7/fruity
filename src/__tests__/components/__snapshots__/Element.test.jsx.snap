// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Element should be rendered correctly. 1`] = `
<div>
  <div
    class="element"
    data-testid="element"
  >
    <label
      class="element__label"
    >
      <input
        class="element__checkbox"
        type="checkbox"
      />
      <img
        alt="apple"
        class="element__image"
        src="apple.webp"
      />
      <span
        class="element__caption"
      >
        APPLE
      </span>
    </label>
  </div>
</div>
`;
